presto-native-execution/presto_cpp/main/SessionProperty.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <optional>
+#include <string>
+#include <typeinfo>
+
+namespace facebook::presto {
+
+// Defines prossible property types.
+enum class PropertyType {
+  kInt,
+  kBool,
+  kLong,
+  kString,
+  // Add more types as needed.
+  kUnknown
+};
+
+/// This is the interface of the session property.
+/// Note: This interface should align with java coordinator.
+class SessionProperty {
```

Comment:
@pramodsatya : Didn't follow why we have separate SessionProperty and SessionPropertyData ? Are we planning to use SessionProperty separately from SessionPropertyData at all ?

Reply:
Thanks for the feedback @aditi-pandit, yes I agree both classes are not needed. I am working on refactoring this PR and the draft PR contains only the cherrypicked commits from fusionnext. I will finish refactoring and open it for review soon.